src/stores/AppStore.js

```diff
@@ -43,6 +44,9 @@ export default {
     this.state.preferredColorScheme = value;
     Settings.preferredColorScheme = value;
   },
+  setAvailableLocales(locales) {
```

Comment:
maybe default to empty array? So we dont fallback all the time when we pass params?

```suggestion
  setAvailableLocales(locales = []) {
```